lang/src/program_account.rs

```diff
@@ -156,6 +157,11 @@ impl<'a, T: AccountSerialize + AccountDeserialize + Clone> Deref for ProgramAcco
 
 impl<'a, T: AccountSerialize + AccountDeserialize + Clone> DerefMut for ProgramAccount<'a, T> {
     fn deref_mut(&mut self) -> &mut Self::Target {
+        if !self.inner.info.is_writable {
```

Comment:
I'm a little hesitant about this change because it introduces an asymmetry between deref and deref_mut and it adds extra runtime checks that are unnecessary--though admittedly it's unlikely this check would create any performance problems.

Is there maybe an alternative to make debugging this situation eaiser? Perhaps better debug logging enabled with the `anchor-debug` feature flag?

Reply:
Sure, I added `anchor-debug` cfg.